src/main/java/com/dataloom/data/DataGraphService.java

```diff
@@ -228,6 +241,57 @@ public void createEntitiesAndAssociations(
                 return Stream.of( writes, addEdge );
             }
         } ).forEach( DataGraphService::tryGetAndLogErrors );
+    }
+
+    @Override
+    public EntitySetData getTopUtilizers(
+            UUID entitySetId,
+            UUID syncId,
+            List<TopUtilizerDetails> topUtilizerDetailsList,
+            int numResults,
+            Map<UUID, PropertyType> authorizedPropertyTypes )
+            throws InterruptedException, ExecutionException {
+        ByteBuffer queryId;
+        try {
+            queryId = ByteBuffer.wrap( ObjectMappers.getSmileMapper().writeValueAsBytes( topUtilizerDetailsList ) );
+        } catch ( JsonProcessingException e1 ) {
+            logger.debug( "Unable to generate query id." );
+            return null;
+        }
+        eds.getEntityKeysForEntitySet( entitySetId, syncId ).parallel().map( entityKey -> {
+            UUID vertexId = idService.getEntityKeyId( entityKey );
+            List<ResultSetFuture> countFutures = new ArrayList<>();
+            for ( TopUtilizerDetails details : topUtilizerDetailsList ) {
```

Comment:
Do a flatMap above and here do `return topUtilizerDetailsList.stream().parallel().map( lm.getEdgeCount(...) )` This will get you a parallel `Stream<ResultSetFuture>`.  Then do `.map( ResultSetFuture::getUninterruptibly )` then do `.map( ResultSet::one )` then do `.mapToDouble( row -> 1.0D*row.getLong(0) )` and finally a `.sum`